apps/api/src/notifications/controllers/notifications.controller.spec.ts

```diff
@@ -548,5 +548,47 @@ describe('NotificationsController', () => {
       expect(result.success).toBe(false);
       expect(result.message).toContain('Error testing SMTP connection');
     });
+
+    it('should use form configuration when provided in DTO', async () => {
```

Comment:
Add a test where if a form value is not provided the db value will still supercede it.

For example, given:
```
      const mockDbConfig = {
        emailEnabled: false,
        smtpHost: 'old-smtp.test.com',
        smtpPort: 25,
        smtpUsername: 'old@example.com',
        smtpPassword: 'oldpassword',
        smtpUseSsl: true,
        senderEmail: 'old-sender@example.com',
        senderName: 'Old Sender',
      };

      const mockFormConfig = {
        emailEnabled: true,
        smtpHost: 'form-smtp.test.com',
        smtpPort: 587,
        smtpUsername: 'form@example.com',
        smtpUseSsl: false,
        senderEmail: 'form-sender@example.com',
        senderName: 'Form Sender',
      };
```
I expect smtpPassword == 'oldpassword'